textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/RegexPredicate.java

```diff
@@ -36,5 +51,36 @@ public boolean satisfy(ITuple tuple) {
         }
         return false;
     }
+    
+    /**
+     * This function return a list of spans in the given tuple that matches the regex 
+     * For example, given tuple ("george watson", "graduate student", 23, "(949)888-8888")
+     * and regex "g[^\s]*", this function will return 
+     * [Span(name, 0, 6, "g[^\s]*", "george watson"), Span(position, 0, 8, "g[^\s]*", "graduate student")]
+     * 
+     * @param tuple document in which search is performed
+     * @return a list of spans describing the occurence of matching sequence in document
+     */
+    public List<Span> statisfySpan(ITuple tuple) {
+    	List<Span> spanList = new ArrayList<>();
+    	if (tuple == null) {
+    		return spanList; //empty array
+    	}
+    	IField field = tuple.getField(fieldName);
+    	if (field instanceof StringField || field instanceof TextField) {
+    		String fieldValue = ((StringField) field).getValue();
+    		if (fieldValue == null) {
+    			return spanList;
+    		} else {
+    			Pattern pattern = Pattern.compile(regex);

```

Comment:
Seems you need to call "compile" and "matcher" for each tuple.  If so, that's very expensive.  Can we construct them once in the constructors then reuse them for each tuple?